File: src/ui/src/shared/types/Creator.ts
import {EntityBase} from "zavadil-ts-common";
import {Account} from "./Account";
import {User} from "./User";

export type CreatorBase = EntityBase & {
	userState: string;
}

export type Creator = CreatorBase & {
	account: Account;
	user: User;
}

export type CreatorStub = CreatorBase & {
	accountId?: number | null;
	userId?: number | null;
}

File: src/main/java/eu/zavadil/openmerch/data/creator/Creator.java
package eu.zavadil.openmerch.data.creator;

import eu.zavadil.openmerch.data.account.Account;
import eu.zavadil.openmerch.data.user.User;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
@Entity
@Table(name = "creator")
public class Creator extends CreatorBase {

	@ManyToOne(optional = false)
	private Account account;

	@ManyToOne(optional = false)
	private User user;
}
